src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,64 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+// TODO the commonalities does not seem to do have anything to do with contact information but rather
+// some kind of validation, perhaps an interface something like Validated might be more suitable.
+// Alternatively, Name actually can also be considered a Contact.
```

Comment:
I think all data field from Java 8 interface is public, so arguably, if you need private data field, an abstract class might be better ^^